File: src/2325-Decode-the-Message/decode_the_message.js
/**
 * @param {string} key
 * @param {string} message
 * @return {string}
 */
var decodeMessage = function(key, message) {
  let sub_table = {}
  let key_index = 0

  for (let i = 0; i < key.length; i++) {
    let c = key[i]

    if (c !== ' ' && !(c in sub_table)) {
      sub_table[c] = String.fromCharCode('a'.charCodeAt(0) + key_index)
      key_index += 1
      
      if (key_index === 26){
        break
      }
    }
  }

  let ans = ''

  for (let i = 0; i < message.length; i++) {
    let c = message[i]

    if (c === ' ') {
      ans += ' '
    }
    else if (c in sub_table) {
      ans += sub_table[c]
    }
  }

  return ans
};

let key = "the quick brown fox jumps over the lazy dog"
let message = "vkbs bs t suepuv"

console.log(decodeMessage(key, message))

File: src/2325-Decode-the-Message/decode_the_message.py
from typing import Dict

class Solution:
    def decodeMessage(self, key: str, message: str) -> str:
        sub_table: Dict[str] = {}
        key_index: int = 0

        for c in key:
            if c != ' ' and c not in sub_table:
                sub_table[c] = chr(ord('a') + key_index)
                key_index += 1
                if key_index == 26:
                    break
        
        ans: str = ''
        for c in message:
            if c == ' ':
                ans += ' '
            elif c in sub_table:
                ans += sub_table[c]
        
        return ans


if __name__ == "__main__":
    key: str = "the quick brown fox jumps over the lazy dog"
    message: str = "vkbs bs t suepuv"
    solution = Solution()
    
    print(solution.decodeMessage(key, message))